File: contrib/matlab/plot_bindata.m
% Reads a binary file and plots a cut in the x-y plane.
% Assumes a simulation in a cubic box with N^3 points.
% Type 'plot_bindata' in MATLAB to execute.

% Path to the data
path = '../../bin/outs/';

% Spatial resolution
N = 128;

% Reads binary files
fid = fopen(strcat(path,'vx.0001.out'));
vx = fread(fid,N*N*N,'float32');
fclose(fid);
vx = reshape(vx,[N N N]);

% Show a horizontal cut of the field in the middle of the box
figure(1)
imagesc(vx(:,:,N/2))
xlabel('x')
ylabel('y')

File: contrib/python/plot_bindata.py
import numpy as np
import matplotlib.pyplot as plt

# Reads a binary file and plots a cut in the x-y plane.
# Execute in ipython with '%run plot_bindata.py'

# Path to the binary data
path = '../../3D/bin/outs/'

# Spatial resolution
NX = 128
NY = 128
NZ = 128
shape = (NX,NY,NZ)

# Reads binary files
vx = np.fromfile(path+'vx.0001.out',dtype=np.float32).reshape(shape,order='F')

# Show a horizontal cut of the field in the middle of the box
plt.figure(1)
plt.imshow(vx[:,:,N/2])
plt.xlabel('x')
plt.ylabel('y')
plt.show()
